### ISSUE TITLE
IMAGE_ONLY mode - stretched output in player

### ISSUE THREAD

**Issue by johnfslattery** [collab-project/videojs-record#511]
## Description
Hello! I've been experimenting with this project (which is a great project!) on a project I've been working on. I'm fairly inexperienced with web development, so please excuse my submission of this issue without an adequate solution. I will try to put one together, but it may take me a long time due to my inexperience.

I have been experimenting with the IMAGE_ONLY mode and my two observations are that:
1) There is no way to specify the camera resolution. In getDevice(), there is a call to navigator.mediaDevices.getUserMedia() in VIDEO_ONLY and AUDIO_VIDEO modes that allows you to pass media constraints. There is no such call for IMAGE_ONLY. I suspect it is just a case of adding it.

2) The rendered image on the canvas when you capture an image stretches to the width/height of the canvas. This is only an issue if the height/width of the camera device differs from the height/width of the player. 

Note: I experienced similar behaviour when using the exportImage() method in AUDIO_VIDEO mode also.

## Steps to reproduce
Use IMAGE_ONLY mode. 
Use a width/height for the player options that differs from the camera width/height.
Optional: Try to pass media constraints. I tried via the video block i.e 
```
record = {
          audio: false,
          video: {
            width:1920,
            height:1080
          },
          frameWidth: 1920,
          frameHeight: 1080,
          debug: true,
          image: true,
          imageOutputType: "dataURL",
          imageOutputFormat: "image/png",
          imageOutputQuality: 0.92,
}
```
Take a capture and observe stretched image.

### Expected
I would expect passing media constraints as an object via image to work as it does when passing them via video in the configuration (as described [here](https://collab-project.github.io/videojs-record/#/media-constraints)).

With regard to the preview of the captured image, I would expect the aspect ratio of the captured image preview to be the same as the camera.

### Actual
Passing media constraints via image or video had no effect on the camera resolution in IMAGE_ONLY mode. 

The preview of the captured image is stretched to the dimensions of the player. 

## Additional Information
Please include any additional information necessary here. Including the following:
#### videojs
4.1.0
#### browsers
I've tried it on Firefox and Chrome (desktop versions for Windows) and Chrome for Android.
#### OSes
I've tried on Windows 10 and Android.

**Comment by thijstriemstra** [comment]
@johnfslattery thanks for the detailed report. It's been a while so doing this from top of my head, but have you tried using `image` instead:

```
record = {
          audio: false,
          image: {
            width:1920,
            height:1080
          },
          frameWidth: 1920,
          frameHeight: 1080,
          debug: true,
          image: true,
          imageOutputType: "dataURL",
          imageOutputFormat: "image/png",
          imageOutputQuality: 0.92,
}
```

**Comment by johnfslattery** [comment]
Hi @thijstriemstra,

Thanks for getting back in touch.

Apologies, I think I've made a mistake in my code. I had image defined twice by accident - once as an object and once as true. Once I removed the latter, the camera resolution is now being read and works well. 

The issue with the image preview is still there though. Could it be that the issue lies when the dimensions of the player and the camera resolution are different? Here is an example using the config below. This config might not work for you if your camera can successfully use the resolution. Another way to demonstrate it might be to create a player larger than the max resolution of your camera, and specify the maximum resolution of the camera so that there are pillarbox bars present. 
 
```
var options = {
    controls: true,
    width: 360,
    height: 640,
    fluid: false,
    bigPlayButton: false,
    controlBar: {
        volumePanel: false,
        fullscreenToggle: false
    },
    plugins: {
        record: {
            image: {
                width: 360,
                height: 640
            },
            frameWidth: 360,
            frameHeight: 640,
            debug: true,
            imageOutputType: 'dataURL',
            imageOutputFormat: 'image/png',
            imageOutputQuality: 0.92
        }
    }
};
```
Before taking snapshot:
![image](https://user-images.githubusercontent.com/9817741/94435596-3c64ee80-0193-11eb-9e47-d1fd6643d003.png)

After taking snapshot:
![image](https://user-images.githubusercontent.com/9817741/94435700-628a8e80-0193-11eb-866b-5cc4e24872ba.png)
